swarm/api/routes/events.py

```
@@ -353,6 +355,11 @@ async def stream_run_events(run_id: str, request: Request):
         event: run:completed
         data: {"run_id": "abc123", "status": "succeeded"}
     """
+    try:
+        run_id = validate_path_component(run_id)
+    except ValueError as e:
+        raise HTTPException(status_code=400, detail=str(e))
```

Ruff (0.14.14)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)